plugins/inputs/prometheus/prometheus.go

```diff
@@ -294,7 +292,10 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 			return fmt.Errorf("unable to create new request '%s': %s", u.URL.String(), err)
 		}
 	}
-
+	acceptHeader := "application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3,*/*;q=0.1"
+	if p.MetricVersion == 2 {
+		acceptHeader = "proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3,*/*;q=0.1"
+	}
```

Comment:
This is a question rather than a request for change: Do you think we can go with
```suggestion
	acceptHeader := "proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3,*/*;q=0.1"
	if p.MetricVersion != 2 {
		acceptHeader = "application/vnd.google.protobuf;" + acceptHeader
	}
```
This way you can directly see the special handling for version 2 which is otherwise hidden in a "long-string-comparison"... Can live with both versions though.